lib/dataset/query.js

```diff
@@ -338,6 +338,50 @@ define({
             return this.mergeOptions({distinct: args});
         },
 
+        /*   {
+         *     division: db.from('division').one();
+         *   }
+         *
+         */
+        eager: function(includeDatasets, model) {
+            this._eager = includeDatasets;
+            this.rowCb = (topLevelResults) => {
```

Comment:
We could copy the original rowCb and invoke it here. Using `when` we could then determine if it returned a model if so use that so you dont have to pass it in.